### PR TITLE
Admin Dashboard about to be ready

### PR DESCRIPTION
**Pull request by sagyMoveo**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import Dashboard from './pages/Dashboard';
+import './App.scss';
+import TryComp from './TryComp';
 
 function App() {
   return (
     <div>
       <Routes>
         <Route path='/' element={<Dashboard />} />
+        <Route path='/try' element={<TryComp />} />
```
Comment (nir-boost) [inline src/App.tsx:11]: 1. We want to redirect when a user puts an unknown URL. 

2. we use all the route paths as const :
const ROUTES ={ 
DASHBOARD: '/dashboard',
TRY: '/try'
}

File: src/TryComp.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { RootState } from './redux/store';
+import { setObjectToUpdateRedux } from './redux/slices/collections.slice';
+
+const TryComp = () => {
+  const dispatch = useDispatch();
+
+  // Retrieve the objectToUpdateRedux value from Redux store
+  const objectToUpdateRedux = useSelector(
+    (state: RootState) => state.collectionsData.objectToUpdateRedux,
+  );
```
Comment (nir-boost) [inline src/TryComp.tsx:11]:
```suggestion
  const { objectToUpdateRedux } = useSelector(
    (state: RootState) => state.collectionsData,
  );
```

File: src/data/utils/emptyObjects/EmptyObjects.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+export const create = (collectionName: string): ICollectionItem => {
+  if (collectionName === 'chefs') {
+    return chefs;
+  }
+  if (collectionName === 'restaurants') {
+    return restaurants;
+  }
+  if (collectionName === 'dishes') {
+    return dishes;
```
Comment (nir-boost) [inline src/data/utils/emptyObjects/EmptyObjects.ts:46]: already have this const? 
if not make one

File: src/redux/slices/collections.slice.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+const initialState: CollectionsDataState = {
+  data: {
+    chefs: [],
+    dishes: [],
+    restaurants: [],
+  },
+  collectionNames: [],
+  currentCollection: 'chefs',
```
Comment (nir-boost) [inline src/redux/slices/collections.slice.ts:20]: use Const

File: src/services/api.service.ts [hunk 5]
```diff
@@ -48,4 +82,61 @@ const getCollectionsNames = async (): Promise<string[]> => {
...
+    throw error;
+  }
+};
+const createOne = async (
+  collectionName: string,
+  objectToCreate: Partial<ICollectionItem>,
+): Promise<ICollectionItem> => {
+  try {
+    console.log('object to create: ', objectToCreate);
+    console.log(`${API_BASE_URL}/${collectionName}`);
```
Comment (nir-boost) [inline src/services/api.service.ts:114]: remove logs

File: src/shared/components/ModelButton/ModelButton.component.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+function ModelButton({ elemId, elemName, collection }) {
+  const handleDeleteTag = async () => {
+    console.log(`Delete element ${elemId} from collection ${collection}`);
```
Comment (nir-boost) [inline src/shared/components/ModelButton/ModelButton.component.tsx:3]: remove log

File: src/shared/components/ModelButton/ModelButton.component.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  };
+  return (
+    <div>
+      {elemName}
+      <button onClick={handleDeleteTag} type='button'>
+        x
```
Comment (nir-boost) [inline src/shared/components/ModelButton/ModelButton.component.tsx:9]: use resources or const

File: src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+      // If the option is already selected, deselect it
+      const index = updatedSelected.indexOf(clickedValue);
+      if (index !== -1) {
+        updatedSelected.splice(index, 1);
+      } else {
+        // If the option is not selected, toggle selection status
+        updatedSelected.push(clickedValue);
+      }
+      console.log('updatedSelected:', updatedSelected);
```
Comment (nir-boost) [inline src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx:33]: remove log

File: src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+      const updatedIds = updatedSelected.map(
+        (option) =>
+          collectionsData.data[collectionName].find(
+            (element: ICollectionItem) => element.name === option,
+          )._id,
+      );
+      console.log({ [fieldName]: updatedIds });
```
Comment (nir-boost) [inline src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx:41]: remove log

File: src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+          <span
+            key={value}
+            onClick={() => handleOptionClick(value)}
+            style={{
+              marginRight: '10px',
+              cursor: 'pointer',
+              backgroundColor: selectedOptions.includes(value)
+                ? 'lightblue'
+                : 'white',
+            }}
```
Comment (nir-boost) [inline src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx:65]: extract style  to style file

File: src/shared/components/Table/Table.component.tsx [hunk 11]
```diff
@@ -1,65 +1,114 @@
...
-          data.map((element: ICollectionItem, rowIndex: number) => (
+    <>
+      <button
+        type='button'
+        className={styles.createNewButton}
+        onClick={() => {
+          setShowCreateLine(!showCreateLine);
+        }}
+      >
+        {/* <span>Create New</span> <img src={addNewIcon} alt='' /> */}
```
Comment (nir-boost) [inline src/shared/components/Table/Table.component.tsx:83]: remove comment

File: src/shared/components/TagsArray/TagsArray.component.tsx [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+function TagsArray({ tagsArray, field }) {
+  const dispatch = useDispatch();
+  const [tags, setTags] = useState(tagsArray);
+  const [tagToAdd, setTagToAdd] = useState('');
+  const [tagsAddedNotSavesYet, setTagsAddedNotSavesYet] = useState([]);
+  const [showAddInput, setShowAddInput] = useState(false);
+  useEffect(() => {
+    console.log('');
+  }, [tags]);
+
```
Comment (nir-boost) [inline src/shared/components/TagsArray/TagsArray.component.tsx:24]: please delete if nor relevant

File: src/shared/components/TagsArray/TagsArray.component.tsx [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+  const handleAddNew = () => {
+    console.log('add new');
```
Comment (nir-boost) [inline src/shared/components/TagsArray/TagsArray.component.tsx:26]: remove logs

File: src/shared/components/TagsArray/TagsArray.component.tsx [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    setTags((prevState) => {
+      const newTags = arrayFunctions.remove(prevState, tagsAddedNotSavesYet);
+      dispatch(updateObjectToUpdateRedux({ [field]: newTags }));
+      return newTags;
+    });
+    setTagsAddedNotSavesYet([]);
+  };
+
+  const handleApply = () => {
+    console.log(tagToAdd);
```
Comment (nir-boost) [inline src/shared/components/TagsArray/TagsArray.component.tsx:42]: remove log

File: src/shared/parsers/TableLine.parser.tsx [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+    return truncatedDescription + (words?.length > 15 ? '...' : '');
+  }
+
+  if (key === OBJECT_KEYS.RESTAURANTS) {
+    if (cell && (cell as []).length > 0) {
+      const names = (cell as [])?.map((item: IRestaurant) => item.name);
+      const joinedNames = names?.join(', ');
+      return joinedNames;
+    }
+    return 'No restaurants assigned';
```
Comment (nir-boost) [inline src/shared/parsers/TableLine.parser.tsx:55]: looks like use the same text and just the model name change please use it in const and resources for it

File: src/shared/parsers/TableLine.parser.tsx [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+        .map((elem: number) => {
+          if (elem === 0) {
+            return 'Spicy';
+          }
+          if (elem === 1) {
+            return 'Vegan';
+          }
+          if (elem === 2) {
+            return 'Vegetarian';
+          }
```
Comment (nir-boost) [inline src/shared/parsers/TableLine.parser.tsx:88]: use const

File: src/shared/parsers/TableLine.parser.tsx [hunk 17]
```diff
@@ -0,0 +1,233 @@
...
+    );
+  }
+
+  if (key === OBJECT_KEYS.RESTAURANT) {
+    return (
+      <ModelSelect
+        currValue={cell[0]?.name}
+        collectionName='restaurants'
+        fieldName='restaurant'
+      />
```
Comment (nir-boost) [inline src/shared/parsers/TableLine.parser.tsx:215]: use const

File: src/shared/parsers/TableLine.parser.tsx [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+import { IRestaurant } from '../../data/interfaces/restaurant.interface';
+import { ICollectionItem } from '../../data/interfaces/general.interface';
+import { IChef } from '../../data/interfaces/chef.interface';
+
+export const renderCell = (cell: ICell, key: string) => {
+  if (key === OBJECT_KEYS.NAME || key === OBJECT_KEYS.ID) {
+    return cell;
+  }
+  if (key === OBJECT_KEYS.IMAGE) {
+    return <img style={{ width: '140px' }} src={cell as string} alt='' />;
```
Comment (nir-boost) [inline src/shared/parsers/TableLine.parser.tsx:22]: extract style  to style file

File: src/shared/components/ImageCell/ImageCell.component.tsx [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+      method: 'post',
+      body: data,
+    })
+      .then((resp) => resp.json())
+      .then((d) => {
+        setUrl(d.url);
+
+        dispatch(updateObjectToUpdateRedux({ image: d.url }));
+      })
+      .catch((err) => console.log(err));
```
Comment (nir-boost) [inline src/shared/components/ImageCell/ImageCell.component.tsx:28]: remove log

File: src/shared/components/TableLine/TableLine.component.tsx [hunk 20]
```diff
@@ -1,32 +1,127 @@
...
+  const objectToUpdateRedux = useSelector(
+    (state: RootState) => state.collectionsData.objectToUpdateRedux,
+  );
+
+  const handleDelete = useCallback(async () => {
+    const deletedObject: ICollectionItem = await ApiService.deleteOne(
+      collectionName,
+      element._id,
+    );
+    console.log(
```
Comment (nir-boost) [inline src/shared/components/TableLine/TableLine.component.tsx:47]: remove log

File: src/services/api.service.ts [hunk 21]
```diff
@@ -48,4 +82,61 @@ const getCollectionsNames = async (): Promise<string[]> => {
...
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(objectToUpdate), // Send the updated object in the request body
+      },
+    );
+    const data: ICollectionItem = await response.json();
+    return data;
+  } catch (error) {
+    console.error('Error updating item:', error);
```
Comment (nir-boost) [inline src/services/api.service.ts:104]: remove log

**Review by nir-boost — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (sagyMoveo) [reply src/App.tsx:11]: done

File: src/TryComp.tsx (hunk 2 above)
Comment (sagyMoveo) [reply src/TryComp.tsx:11]: done

File: src/data/utils/emptyObjects/EmptyObjects.ts (hunk 3 above)
Comment (sagyMoveo) [reply src/data/utils/emptyObjects/EmptyObjects.ts:46]: done

File: src/redux/slices/collections.slice.ts (hunk 4 above)
Comment (sagyMoveo) [reply src/redux/slices/collections.slice.ts:20]: done

File: src/shared/components/ModelButton/ModelButton.component.tsx (hunk 6 above)
Comment (sagyMoveo) [reply src/shared/components/ModelButton/ModelButton.component.tsx:3]: Component deleted.

File: src/shared/components/ModelButton/ModelButton.component.tsx (hunk 7 above)
Comment (sagyMoveo) [reply src/shared/components/ModelButton/ModelButton.component.tsx:9]: Component deleted.

File: src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx (hunk 8 above)
Comment (sagyMoveo) [reply src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx:33]: done

File: src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx (hunk 9 above)
Comment (sagyMoveo) [reply src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx:41]: done

File: src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx (hunk 10 above)
Comment (sagyMoveo) [reply src/shared/components/ModelSelectMultiple/ModelSelectMultiple.component.tsx:65]: done

File: src/shared/components/TagsArray/TagsArray.component.tsx (hunk 13 above)
Comment (sagyMoveo) [reply src/shared/components/TagsArray/TagsArray.component.tsx:26]: done

File: src/shared/parsers/TableLine.parser.tsx (hunk 16 above)
Comment (sagyMoveo) [reply src/shared/parsers/TableLine.parser.tsx:88]: done.

File: src/shared/parsers/TableLine.parser.tsx (hunk 17 above)
Comment (sagyMoveo) [reply src/shared/parsers/TableLine.parser.tsx:215]: done.

File: src/shared/parsers/TableLine.parser.tsx (hunk 18 above)
Comment (sagyMoveo) [reply src/shared/parsers/TableLine.parser.tsx:22]: done

File: src/shared/components/ImageCell/ImageCell.component.tsx (hunk 19 above)
Comment (sagyMoveo) [reply src/shared/components/ImageCell/ImageCell.component.tsx:28]: changed to console.error

File: src/shared/components/TableLine/TableLine.component.tsx (hunk 20 above)
Comment (sagyMoveo) [reply src/shared/components/TableLine/TableLine.component.tsx:47]: done

File: src/services/api.service.ts (hunk 21 above)
Comment (sagyMoveo) [reply src/services/api.service.ts:104]: can i let it stay as console.error?
or its better be off no matter what?

**Review by nir-boost — APPROVED** [review]

**Pull request merged**